ext/hal/st/stm32cube_subg1/README

```diff
@@ -0,0 +1,36 @@
+STM32CubeSUBG1
+##############
+
+Origin:
+   ST Microelectronics
+   https://www.st.com/content/st_com/en/products/embedded-software/mcu-mpu-embedded-software/stm32-embedded-software/stm32cube-expansion-packages/x-cube-subg1.html
+
+Status:
+   version 3.1.0
+
+Purpose:
+   ST Microelectronics official software package for SUBG1 radio series.
+
+Description:
+   This package is an extract of the official STM32CubeSUBG1 expansion package
+   written by ST Microelectronics. It contains the software library for spirit1 and
+   s2-lp 802.15.4 radios
+
+Dependencies:
+    None.
+
+URL:
+   https://www.st.com/content/st_com/en/products/embedded-software/mcu-mpu-embedded-software/stm32-embedded-software/stm32cube-expansion-packages/x-cube-subg1.html
+
+commit:
+   version 3.1.0
+
+Maintained-by:
+   External
+
+License:
+   BSD-3-Clause
```

Comment:
This would be great, but where did you found the info?
Looking to proposed file, I can't find this clearly stated

Reply:
This is a placeholder that I forgot to remove, when I copied the format of the other st-cube libraries (e.g. stm32f0xx)

Reply:
The actual licence for this lib is this: https://my.st.com/content/ccc/resource/legal/legal_agreement/license_agreement/group0/62/6a/48/15/b4/12/47/7a/DM00367782/files/DM00367782.pdf/jcr:content/translations/en.DM00367782.pdf

Reply:
For reference you can find the licence link here: https://my.st.com/content/my_st_com/en/products/embedded-software/mcu-mpu-embedded-software/stm32-embedded-software/stm32cube-expansion-packages/x-cube-subg1.license=1557757423596.product=X-CUBE-SUBG1.version=3.1.0.html#

Reply:
Ok. Well this is actually a License agreement, and not exactly a License.
It says that we should use components of the package according to the License they disclaim.
So either this is II and we're good:
- FOR THE LICENSED SOFTWARE OR ANY PART THEREOF WHICH CONTAINS OPEN SOURCESOFTWARE: THE APPLICABLE OPEN SOURCE TERMS SHALL APPLY.

Either this is III, and we have an issue:
-  THE LICENSED SOFTWARE OR ANY PARTS THEREOF MAY (I) REQUIRE LICENSES FROMTHIRD PARTIES CLAIMING INTELLECTUAL PROPERTY RIGHTS COVERING USE ORIMPLEMENTATION OF THE LICENSED SOFTWARE OR (II) BE BASED ON INDUSTRYRECOGNIZED STANDARDS OR SOFTWARE PROGRAMS PUBLISHED BY INDUSTRYRECOGNIZED STANDARDS BODIES AND CERTAIN THIRD PARTIES MAY CLAIM TO OWNINTELLECTUAL PROPERTY RIGHTS THAT COVER IMPLEMENTATION OR USE OF THOSESTANDARDS. YOU AGREE THAT YOU ARE RESPONSIBLE FOR OBTAINING ANY SUCH LICENSEWHICH MAY BE NEEDED, AND NO SUCH LICENSE IS PROVIDED BY ST OR ITS AFFILIATES TOYOU .SUCH THIRD PARTY INTELLECTUAL PROPERTY RIGHTS ARE NOT SUBJECT TO THETERMS OF THIS AGREEMENT TO THE EXTENT THE TERMS OF THIS AGREEMENT ARE INCONFLICT WITH SUCH APPLICABLE THIRD PARTY INTELLECTUAL PROPERTY RIGHTS.

So we need to find out which one applies here.

Reply:
Is there something I can do to help with this?

Reply:
I don't think so. I'll check internally and will keep you posted.
It might take a while, so review should continue on pure code part zephyr integration during this time.

Reply:
I have the confirmation that ST proposes this library in a permissive License (BSD-3 Clause or alike). So there is nothing that prevent its use in Zephyr.
We just need to get an updated version of the lib with updated header files reflecting this.